### ISSUE TITLE
Results for month if eventnews is used and event duration is f.e. from 25 August to 4. September

### ISSUE THREAD

**Issue by chequille** [georgringer/news#1816]
Hi Everybody,

I think I asked this a long time ago, but could not find my issue.

Therefore, I am making this new one.

I am using news and eventnews, currently with the latest versions for Typo3 8.x.x

I have events which are starting in one month and end in the next month, f.e. BeginDate 25.08. and EndDate 04.09.

My problem is, that such an event is only shown when selecting August in my Month selection but not when selecting September.

Is there some solution to this or is it implemented in the current versions of news and eventnews?
I need to upgrade my Typo3 to 10 or 11 soon, therefore it would be OK, if this would work with the current versions.

Looking forward to hearing from you.

Best regards,
Jürgen

**Comment by georgringer** [comment]
I suggest to please test this in latest versions yourself. Just set up a TYPO3 11 with latest news and latest eventnews. 

Thanks for checking and giving feedback

**Comment by chequille** [comment]
Hmm ???
It is a way to see if the latest version doing it different than the one I am using.
But would it no be much easier to have a look at the SQL query which is getting the news collection when a dedicated month is selected?
I do not want you to fix this for the version I am using, but maybe you can give me a hint where I can have a look into the code to maybe change it for my purpose.
I guess there is some where clause in the query like (startdate >= firstofthemonth and startdate <= lastofthemonth) 
this must be enhanced with something like  "or (enddate >= firstofthemonth and enddate <= lastofthemonth)

If you can give me an idea in which module or part of the extension I can find the collection query it would help me to maybe get it working as I want.

Best regards,
Chequille

**Comment by georgringer** [comment]
https://github.com/georgringer/eventnews/blob/main/Classes/Hooks/AbstractDemandedRepository.php#L139-L161 should be the related code

**Comment by chequille** [comment]
Hi,
I tested now with Typo3 version 10.4.22 and news 9.4.0 and eventnews 5.0.0 and it is not working.
Event with BeginDate 25.08. and EndDate 04.09. is not shown in September
What I found is, that in the above mentioned AbstractDemandedRepository class from eventnews, $demand->getMonth() and $demand->getYear() is always 0, it does not contain the value from the command. Therefore it cannot work in any case.
As well there is another problem: in this class, override of the original demand ony happens if eventrestrictions is set to EVENT_RESTRICTION_ONLY_EVENTS, but i do have both, normal news and event news.
I set this eventrestriction but no change.

So at the moment it seems for me that there is no working solution. Maybe I have to do the code change to get a right query direct in NewsRepository class. Any hint for doing this?

Regards,
Chequille

**Comment by chequille** [comment]
Hi Georg,
I have some new information.
I installed eventnews within Typo3 from TER and there it is version 5.0.0.
But here in github, you have a version 5.0.1
Tried this version by downloding to my PC and uploading the files to my Typo3 10.x.x setup and it is working :-)
My example event is shown in August and in September as I wanted to have it.

So, the change between eventnews 5.0.0 and 5.0.1 solved the problem.

Best regards,
Chequille
